### PR TITLE
[KYUUBI #3875] Filter out kyuubi.metadata.store.* configurations in e…

### PR DESCRIPTION
**Pull request by lightning-L**
…ngine process builder

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

Before modification:
snipaste_2022-11-29_21-16-14

After modification:
Snipaste_2022-11-29_22-20-11

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala [hunk 1]
```diff
@@ -182,6 +182,10 @@ private[kyuubi] class EngineRef(
 
     conf.set(HA_NAMESPACE, engineSpace)
     conf.set(HA_ENGINE_REF_ID, engineRefId)
+
+    val metadataStoreConf = conf.getAll.filter(x => x._1.startsWith(KYUUBI_METADATA_STORE_PREFIX))
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:186]: I suppose we can add those configurations to `KyuubiConf#serverOnlyConfEntries` to solve this problems.

Comment (lightning-L) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:186]: The metadata sore related configs are defined in `org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStoreConf._`, which is in kyuubi-server. KyuubConf is in kyuubi-common and has no access to JDBCMetadataStoreConf. Any solution for this?

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:186]: Emm... how about
1. making `serverOnlyConfEntries` mutable
2. adding method `KyuubiConf#registerServerOnlyConfEntry`
3. invoking `KyuubiConf#registerServerOnlyConfEntry` in `JDBCMetadataStoreConf`

Comment (lightning-L) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:186]: what about making `serverOnly` a property for `ConfigEntry` and `ConfigBuilder`? just like `internal`

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:186]: sgtm

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -2153,7 +2156,7 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(true)
 
-  private val serverOnlyConfEntries: Set[ConfigEntry[_]] = Set(
+  private var serverOnlyConfEntries: Set[ConfigEntry[_]] = Set(
     FRONTEND_BIND_HOST,
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2160]: can we update these config entries as well?

**Review by pan3793 — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**